client.go

```diff
@@ -70,26 +78,40 @@ type Client struct {
 }
 
 // NewClient returns a new API Client using the given APIKey
-func NewClient(apiKey string) *Client {
+func NewClient(apiKey string, options ...ClientOptions) *Client {
+	host := APIHost
+	for _, option := range options {
+		if option.region == "eu" {
+			host = EuAPIHost
+		}
+	}
+
 	return &Client{
 		apiKey:     apiKey,
-		baseURL:    APIHost,
+		baseURL:    host,
 		Log:        NewLogger(LevelWarn),
 		HTTPClient: defaultClient,
 	}
 }
 
 // newClient creates a new Recurly API Client
-func newClient(apiKey string, httpClient *http.Client) *Client {
+func newClient(apiKey string, httpClient *http.Client, options ...ClientOptions) *Client {
```

Comment:
Given that this is an internal package function, it's fine to change this signature. However, it's a little awkward that it duplicates the logic of the public function. Ideally the public function would call this private function and the logic would exist only once.

For now it's okay to leave as is since it's just a couple lines of code; feel free to reach out if you'd like help with that refactor though.

Reply:
if this part of code is only used for specs and client isn't using it, maybe we could change the signature to use ClientOptions and move httpClient to it. with that in mind this could be the second newClient signature instead of NewClientWithOptions. but I don't think that we have the information if client use this func.